src/main/java/com/r307/arbitrader/service/model/ExchangeFee.java

```diff
@@ -0,0 +1,38 @@
+package com.r307.arbitrader.service.model;
+
+import java.math.BigDecimal;
+import java.util.Optional;
+
+public class ExchangeFee {
+    private BigDecimal shortFee;
+    private BigDecimal longFee;
+
+    public ExchangeFee(BigDecimal shortFee, BigDecimal longFee) {
+        this.shortFee = shortFee;
+        this.longFee = longFee;
+    }
+
+    public Optional<BigDecimal> getShortFee() {
+        return Optional.ofNullable(shortFee);
+    }
+
+    public void setShortFee(BigDecimal shortFee) {
+        this.shortFee = shortFee;
+    }
+
+    public BigDecimal getLongFee() {
```

Comment:
Not super important but from a defensive code perspective this strikes me as a little strange, that the short one has `Optional` and this one doesn't. There isn't any protection in the setter or the constructor against setting this to `null`. I know it should never be null because there's always a fee, but it's totally possible to set it null by accident or malice and crash the program.

This object seems like a good candidate to be immutable anyway since the fees don't change after the instance is created. Then it would make more sense to not have `Optional` because there's no way you could set it null, and you could throw an exception if you try to make one with a null `longFee`.

Reply:
I agree with the object being immutable and I am usually all the way for immutable objects. I don't remember why I did not set it immutable neither I see a good reason to not be immutable so I will make the fields final.

The `public Optional<BigDecimal> getShortFee()` is optional because not all exchanges are margin exchanges so for some exchanges if we call `getShortFee()` we may actually get a null object and if we don't do a null check then we will get a NPE. Making `getShortFee()` return an `Optional` is a sign that this value may be null and it will force the developer to take that in consideration.

`getLongFee()` is does not return `Optional` because it should never be null. If it is null then the program should crash. 
If I got you right, you would also let the program crash if `shortFee` is also null?

Reply:
Don't get me wrong, I'm all for using `Optional`. I just have an irrational need for things to be symmetrical. 😂

If there's no reason _not_ to make it immutable, let's do that. Then there wouldn't be a way to make `getLongFee()` return null (by setting it null after construction or passing it into the constructor) and break the contract of the object. Let's also make it blow up if you pass it a null `longFee` through the constructor, or at least annotate it `@NotNull` so the IDE warns you not to do that.

Reply:
I changed the constructor to include `@NotNull BigDecimal tradeFee` and inside the constructor I added `Objects.requireNonNull(tradeFee);`

Regarding the `Optional`, are you fine with?